### PR TITLE
Print file parameter

### PR DESCRIPTION
**Pull request by sapir**
for #219

### CODE REVIEW

File: vm/src/builtins.rs [hunk 1]
```diff
@@ -559,32 +559,60 @@ pub struct PrintOptions {
...
-        } else {
-            write!(stdout_lock, " ").unwrap();
+    if let Some(file) = options.file {
+        let sep = match options.sep {
+            Some(sep) => sep.into_pyobject(vm).unwrap(),
+            _ => PyString::from(" ").into_pyobject(vm).unwrap(),
+        };
+
+        let mut first = true;
+        for object in objects {
```
Comment (windelbouwman) [inline vm/src/builtins.rs]: This for loop is now duplicated, including the `first` variable. I would change this into something like this:

Enter pseudo rust code:
```rust

// Create some sort of closure function:
let write_func = if let Some(file) = options.file {
  get_
} else {
  func_with_write_macro
}

// From this point, we have the availability over a function which does the proper printing, and can use this function at will.

// Now loop over the objects and print them using the closure function:
let mut first = true;
for object in objects {
  
  if first {
    first = false;
  } else {
    write_func(sep.clone())
  }
  write_func(object)
}
```

Comment (windelbouwman) [reply vm/src/builtins.rs]: Additional information: https://doc.rust-lang.org/book/ch13-01-closures.html

Comment (sapir) [reply vm/src/builtins.rs]: better now?

Comment (windelbouwman) [reply vm/src/builtins.rs]: Nice job!

**Pull request merged**